spec/support/helpers/journey_assertion_helper.rb

```diff
@@ -0,0 +1,45 @@
+module Helpers
+  module JourneyAssertionHelper
+    def navigate_to_page(path, submit_form: false, submit_button_text: "Continue", axe_check: true, &block)
+      visit(path)
+
+      expect(page.current_path).to eql(path)
+
+      page_checks(axe_check:, &block)
+
+      page.click_button(submit_button_text) if submit_form
+    end
+
+    def given_i_am_on_page(path, submit_form: true, submit_button_text: "Continue", axe_check: true, &block)
+      expect(page.current_path).to eql(path)
+
+      page_checks(axe_check:, &block)
+
+      page.click_button(submit_button_text) if submit_form
+    end
+
+    alias_method :now_i_should_be_on_page, :given_i_am_on_page
```

Comment:
Perhaps `now_i_should_be_on_page` could be better named - it doesn't really explain that we also will be ensuring the page is Axe valid and that we'll be submitting it. Couldn't come up with anything though.

Reply:
The three of these don't fully encapsulate what is going on IMO, it's unclear that navigate_to_page will include expectations or execute a block. 

Given that navigate_to_page is mostly duplicating logic from given_i_am_on_page, I'd pitch removing the page_checks and submission parts of that method and instead expect the tester to call given_i_am_on_page.

I'd also propose updating the names of the latter two methods to use language that is more idiomatic of rspec+capybara, for example:
- given_i_am_on_page -> expect_page_to_have (and turn path into a keyword parameter)
- and_the_check_your_answers_page_should_contain -> expect_check_answers_page_to_have_answers

Example calls:
```
expect_page_to_have(path: "/registration/choose-school", submit_form: true) do
  expect(page).to have_text("Choose your workplace")

  within ".npq-js-reveal" do
    page.fill_in "Enter the name of your workplace", with: "open"
  end

  expect(page).to have_content("open welsh school")

  page.find("#school-picker__option--0").click
end
```
Note that I've made submit_form false by default here so that you have to be explicit about taking an action, rather than have to suppress the behaviour.

```
expect_check_answers_page_to_have_answers(
  "cohort" => 2022,
  "course_id" => Course.find_by_code(code: :NPQSL).id,
  ...
)
```